addons/flannel/template/base/install.sh

```diff
@@ -307,20 +307,72 @@ function flannel_pods_remain() {
 }
 
 function remove_weave() {
-    # firstnode only
-    kubectl -n kube-system delete daemonset weave-net
-    kubectl -n kube-system delete rolebinding weave-net
-    kubectl -n kube-system delete role weave-net
-    kubectl delete clusterrolebinding weave-net
-    kubectl delete clusterrole weave-net
-    kubectl -n kube-system delete serviceaccount weave-net
-    kubectl -n kube-system delete secret weave-passwd
+    local resources=("daemonset.apps/weave-net"
+                     "rolebinding.rbac.authorization.k8s.io/weave-net"
+                     "role.rbac.authorization.k8s.io/weave-net"
+                     "clusterrolebinding.rbac.authorization.k8s.io/weave-net"
+                     "clusterrole.rbac.authorization.k8s.io/weave-net"
+                     "serviceaccount/weave-net"
+                     "secret/weave-passwd")
+
+    for resource in "${resources[@]}"; do
+        if ! delete_resource_gracefully "kube-system" "$resource"; then
+            logWarn "$resource is still present after the deletion attempt with a grace period of 1 minute."
+        fi
+    done
 
     # all nodes
     ip link delete weave
+
     rm -rf /var/lib/weave
     rm -rf /etc/cni/net.d/*weave*
     rm -rf /opt/cni/bin/weave*
+
+    logStep "Verifying Weave removal"
+
+    log "waiting up to 1 minute for weave resources to be deleted"
+    if ! spinner_until 60 resource_exists "kube-system" "${resources[@]}"; then
+        logWarn "Weave resources are still present after grace period. Force deletion"
+
+        for resource in "${resources[@]}"; do
+            if ! force_delete_resource "kube-system" "$resource"; then
+                logWarn "Failed to force delete resource: $resource within the timeout period"
+            fi
+        done
+
+        echo "waiting 30 seconds to check removal"
+        sleep 30
+
+        for resource in "${resources[@]}"; do
+            if resource_exists "kube-system" "$resource"; then
+                logWarn "Unable to remove resource: $resource"
+                return
+            fi
+        done
+    fi
+
+    logSuccess "Weave has been successfully removed."
+}
+
+# To delete resources with grace period using timeout
+function delete_resource_gracefully() {
+    local namespace="$1"
+    local resource="$2"
+    timeout 70 kubectl -n "$namespace" delete "$resource" --ignore-not-found --grace-period=60 >/dev/null 2>&1
```

Comment:
i feel like suppressing output that you do not expect to fail is a bad choice

```suggestion
    timeout 70 kubectl -n "$namespace" delete "$resource" --ignore-not-found --grace-period=60 1>/dev/null
```